README.md

```diff
@@ -1,46 +1,177 @@
+      return 'x';
+    } catch (error) {
+      this.logger.error(error, 'Caught a thing!');
+      throw error;
+    }
   }
 }
+
+const x: ClassX = new ClassX();
+
+const doAThing = async () => {
+  x.doSync('Hello Viewer');
+  x.doASync('Hello Longer Guy');
+};
+
+doAThing();
+
 ```
 
-Once you have your logger you can use the functions within and proper gating of execution will be done based on the log level or conditions provided.
+## Decorators
+
+We have two decorators that can help provide a bit more traceability to your logging, `@inOut` and `@duration`.
+
+### @inOut({level, name})
+
+This decorator can be placed on a function to record each time it is executed. By default it logs at trace level, as the amount of logging can be quite verbose and detailed. 
+
+##### Synchronous Example
 
-# Log Level's
+```
+{
+  at: 2019-12-07T20:17:41.992Z,
+  clazz: 'ClassX',
+  func: 'doSync()',
+  msg: 'inOut -->',
+  data: { args: [ 'Hello Viewer' ] },
+  name: 'ClassX.inOut',
+  level: 'debug'
+}
+{
+  at: 2019-12-07T20:17:42.005Z,
+  clazz: 'ClassX',
+  func: 'doSync()',
+  msg: 'inOut <--',
+  data: { result: 'x' },
+  name: 'ClassX.inOut',
+  level: 'debug'
+}
+```
+
+##### Asynchronous Functions
 
-Your logging level will determine what statments are printed out. It is inclusive of the level you set, and all 'higher' level statements. The levels are trace, debug, info, warn and error. So if you set your level to trace, then all messages will be output. If you set your level to warn, only warn and error level statements will be output.
+In the case of async functions, you will see 3 output statements as an additional one is added to record the fact the method returned a Promise, but did not yet resolve.
 
-# Environment Variables
+```
+{
+  at: 2019-12-07T20:17:42.007Z,
+  clazz: 'ClassX',
+  func: 'doASync()',
+  msg: 'inOut -->',
+  data: { args: [ 'Hello Longer Guy' ] },
+  name: 'ClassX.inOut',
+  level: 'debug'
+}
+{
+  at: 2019-12-07T20:17:42.013Z,
+  clazz: 'ClassX',
+  func: 'doASync()',
+  msg: 'inOut -- promisified function',
+  data: { result: {} },
+  name: 'ClassX.inOut',
+  level: 'debug'
+}
+{
+  at: 2019-12-07T20:17:42.016Z,
+  clazz: 'ClassX',
+  func: 'doASync()',
+  msg: 'inOut <--',
+  data: { result: 'x' },
+  name: 'ClassX.inOut',
+  level: 'debug'
+}
+```
 
-You can define default log levels for namespaces as well as a global default using the `LOG_LEVEL` environment variable. Below are the rules to adhere to:
+### @duration({level, name})
 
-1. Log levels are integers 0 = trace, 3 = debug, 5 = info (default), 7 = warn, 10 = error
-1. For multiple values, separate each definition with a semi-colon ';'.
-1. To set the level for a specific group or namespace use the name first and the log level second.
+So this decorator will tell you the amount of time it takes for your function to execute. If this is an async function, it will be the amount of time util the promise resolves, not the amount of time to return the promise.
```

Comment:
```suggestion
This decorator will tell you the amount of time it takes for your function to execute. If this is an async function, it will be the amount of time until the promise resolves, not the amount of time to return the promise.
```